crates/wit-parser/src/resolve.rs

```diff
@@ -1474,22 +1490,26 @@ impl Remap {
     fn append(
         &mut self,
         resolve: &mut Resolve,
-        unresolved: UnresolvedPackage,
+        unresolved: &mut UnresolvedPackage,
     ) -> Result<PackageId> {
         self.process_foreign_deps(resolve, &unresolved)?;
 
         let foreign_types = self.types.len();
         let foreign_interfaces = self.interfaces.len();
         let foreign_worlds = self.worlds.len();
 
-        let pkgid = resolve.packages.alloc(Package {
-            name: unresolved.name.clone(),
-            docs: unresolved.docs.clone(),
-            interfaces: Default::default(),
-            worlds: Default::default(),
-        });
-        let prev = resolve.package_names.insert(unresolved.name.clone(), pkgid);
-        assert!(prev.is_none());
+        let pkgid = if resolve.package_names.contains_key(&unresolved.name) {
+            resolve.package_names.get(&unresolved.name).unwrap().clone()
+        } else {
+            let pkgid = resolve.packages.alloc(Package {
+                name: unresolved.name.clone(),
+                docs: unresolved.docs.clone(),
+                interfaces: Default::default(),
+                worlds: Default::default(),
+            });
+            resolve.package_names.insert(unresolved.name.clone(), pkgid);
+            pkgid
+        };
```

Comment:
As I read over this I don't think that this is quite right, but not because of this PR. This is actually a preexisting issue today. Basically this isn't quite right because below the packages need to be merged rather than appended to otherwise duplicates might show up.

That being said I'll try to fix this in a follow up since it's a preexisting issue independent of this PR.